upup/pkg/kutil/rollingupdate_cluster_validate_drain.go

```diff
@@ -0,0 +1,319 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kutil
+
+// TODO move this business logic into a service than can be called via the api
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/golang/glog"
+	api "k8s.io/kops/pkg/apis/kops"
+	validate "k8s.io/kops/pkg/validation"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awsup"
+	k8s_clientset "k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+)
+
+// RollingUpdateCluster restarts cluster nodes
+type RollingUpdateClusterDV struct {
```

Comment:
That was part of the refactor, to use the feature flag.  I can put the files in separate folders, or actually spell.  I think I will refactor and spell `DrainAndValidate` :)